model/src/main/scala/cromwell/pipeline/model/validator/WrappedFactoryPattern.scala

```diff
@@ -0,0 +1,21 @@
+package cromwell.pipeline.model.validator
+
+package cromwell.pipeline.model.validator
+
+import cats.Show
+import cats.data.Validated.{ Invalid, Valid }
+import cats.data.{ NonEmptyChain, Validated }
+
+trait WrappedFactoryPattern[Type, Error, Wrapper <: Wrapped[Type]] {
+  type ValidationResult[A] = Validated[NonEmptyChain[Error], A]
+  implicit def wrappedOrdering(implicit ord: Ordering[Type]): Ordering[Wrapper] = Ordering.by(_.unwrap)
+  def create(value: Type): Wrapper
+  def validate(value: Type): ValidationResult[Type]
+  def from(value: Type): ValidationResult[Wrapper] =
+    validate(value).map(create)
+  final def apply(value: Type, evidence: Enable.Unsafe.type)(implicit show: Show[Error]): Wrapper =
+    validate(value) match {
+      case Valid(content)  => create(content)
+      case Invalid(errors) => throw new Enable.UnsafeException(errors.toNonEmptyList)
+    }
+}
```

Comment:
I assume there is a reason to factor out this trait from a companion object. What is this reason? :)

Reply:
The reason is that it leads to a misunderstanding of the whole mechanism of how microformatters work. I spent a considerable amount of time to learn that it was no reason to place this in companion trait. This code feature distracts attention from more meaningful details. So, I share position, - "the less complexity - the better choice". And if no need in constructing additional abstractions, we better would avoid them, don't we?

Reply:
> the less complexity - the better choice

Can't argue with that. But I don't understand, how it simplifies the existing code. Why do you find it distracting?
Perhaps there was no reason to place this trait in the companion object. But it does mean that there is a reason to put it not in a companion object.
I do not insist on leaving the existing code without changes. I'm just trying to understand what is going on.

Reply:
I thought that placing logic to the companion object is needed to share namespace between universal trait and it. But this trait plays only one role - it needs to make factories of microtypes, it's their supertype. It doesn't depend on a universal trait, and universal trait doesn't depend on it. They are neither share any sense of code nor any kind of logic. Do we wire anything if there is no reason to do it? I looked for the reason, but I did not find anything. While students look for it, thy, I suppose to find it scary, and they drop their attempts to understand it. That is why I decided to do it. If, of course, at least, one reason to do it exist, I am sure, I have to leave it as it was and don't touch it.

Reply:
I can't agree that they do not share logic.
Microtypes without companion objects useless since we don't "know" how to create them. @ilyaonishenko what do you think?

Reply:
yes, but companions such as Name, Password, etc are not the same as WrappedFactory/Wrapped companion

Reply:
`object Password extends WrappedFactoryPattern[String, String, Password]`
Honestly, they `Password` looks the same as `WrappedFactory` to me.

Reply:
renamed this to WrappedFactory. Is it satisfied?

Reply:
I'm sorry if my typo confused you.
There is nothing wrong with `WrappedFactoryPattern` or `WrappedFactory` name. For now, I don't care about the name, I'm more concerned about the main idea.

Reply:
done